draft-ietf-ohai-ohttp.md

```diff
@@ -974,12 +976,33 @@ examine the content of a request (other than to observe the choice of key
 identifier, KDF, and AEAD), so it is also assumed that it cannot generate an
 encapsulated response.
 
-A proxy MUST NOT add information about the client identity when forwarding
-requests. This includes the Via field, the Forwarded field
-{{?FORWARDED=RFC7239}}, and any similar information.  A client does not depend
-on the proxy using an authenticated and encrypted connection to the oblivious
-request resource, only that information about the client not be attached to
-forwarded requests.
+A proxy SHOULD NOT add information about the client identity when forwarding
+requests. For example, including the Via field, the Forwarded field
+{{?FORWARDED=RFC7239}}, or similar information may reveal information about the
+client that the proxy is intended to hide from the target.  In general, a client
+does not depend on the proxy using an authenticated and encrypted connection to
+the oblivious request resource, only that information about the client not be
+attached to forwarded requests. However, some applications or use cases can
+benefit additional client metadata attached to forwarded request. Some example
+HTTP headers that the proxy might include are as follows:
+
+- Flag indicating whether or not the client is deemed malicious. If a client
+  request orignates from an abusive or otherwise harmful IP address with a bad
+  reputation, the proxy might attached a flag indicating this to the target. The
+  target could use this flag to shadowban clients or other alter whether how
+  the client encapuslated request is processed.
+- Convey coarse geographic information about the client. Some applications, such
+  as telemetry, may benefit from approximate geographic information about a
+  client request. Clients could of course provide such information in their
+  encapsulated request, but proxies ultimately have a more authoritative view
+  of the client request origin and its geographic location by virtue of being
+  on path between the client and target. As such, for these applications, it
+  would be reasonable for proxies to supply coarse geographic information about
+  individual client requests.
+
+Servers that enter into an agreement with a proxy are expected to agree upon the
+types of client metadata that proxies provide, if any. Clients need to consider
+this information when choosing a proxy discovery method.
```

Comment:
From the perspective of a client, this text horrifies me.

It says that servers and proxies reach an agreement and then clients take it or leave it.  That's a terrible negotiating position.  I know it's probably how it will work, but it's not healthy.

Reply:
The approach we’ve been taking is that the client is always the one to attach information in the encrypted payload to the next hop, which it may learn from the proxy, and reflect after deciding what it is OK sending. But generally it’s not OK to let the proxy attach any information without the client’s consent. The marker for “bad client” shadow bans seems to be the exception case?

Reply:
There's two types of info the proxy might attach: info that the client agrees to have attached, like geo or whatever, and info that the client does not have consent to attach. Certainly clients shouldn't use a proxy if it attaches known info they don't agree to. But they can't really do anything about the former. So maybe the best outcome is (1) here is that clients agree to known info that the proxy attaches, and (2) proxies MUST NOT attach more than a bit (or two or whatever is a reasonable upper bound) for info that the clients explicitly do not agree to?

Reply:
FWIW, I made this change.